WcaOnRails/app/models/competition.rb

```diff
@@ -1164,30 +1165,47 @@ def persons_wcif
   end
 
   def events_wcif
-    competition_events.map(&:to_wcif)
+    includes_associations = [
+      { rounds: [:competition_event, :wcif_extensions] },
+      :wcif_extensions,
+    ]
+    competition_events.includes(includes_associations).map(&:to_wcif)
   end
 
   def schedule_wcif
+    competition_venues_includes_associations = [
+      {
+        venue_rooms: [
+          :wcif_extensions,
+          { schedule_activities: [{ child_activities: [:child_activities, :wcif_extensions] }, :wcif_extensions] },
+        ],
+      },
+      :wcif_extensions,
+    ]
     {
       "startDate" => start_date.to_s,
       "numberOfDays" => number_of_days,
-      "venues" => competition_venues.map(&:to_wcif),
+      "venues" => competition_venues.includes(competition_venues_includes_associations).map(&:to_wcif),
     }
   end
 
   def set_wcif!(wcif, current_user)
     JSON::Validator.validate!(Competition.wcif_json_schema, wcif)
     ActiveRecord::Base.transaction do
-      update_persons_wcif!(wcif["persons"], current_user) if wcif["persons"]
       set_wcif_events!(wcif["events"], current_user) if wcif["events"]
       set_wcif_schedule!(wcif["schedule"], current_user) if wcif["schedule"]
+      update_persons_wcif!(wcif["persons"], current_user) if wcif["persons"]
```

Comment:
Out of curiosity, why the reordering here? If there is an important reason, it would be nice to make sure we have a test that ensures we don't regress here.

EDIT: I finished reading the diff, I think this is covered by the `allows adding assignments for newly added activities` test. Thanks for writing that!

Reply:
Yeah, I started with the test to make sure it breaks ^^